hepcrawl/extractors/iop_parser.py

```diff
@@ -124,3 +120,47 @@ def get_arxiv_eprints(self, node):
             arxiv_eprints.append({'value': pattern.sub('', arxiv_value)})
 
         return arxiv_eprints
+
+    def get_license_type_and_version_from_url(self, url):
+        LICENSE_PATTERN = re.compile(r"([A-Za-z]+)\/([0-9]+\.[0-9]+)")
+        CREATIVE_COMMONS_PATTERN = re.compile(r".*(creative\s{0,}commons).*", flags=re.I)
+        match = LICENSE_PATTERN.search(url)
+        if not match:
+            print("No license found in URL")
+            return None
+        first_part_of_license_type = ""
+        version = match.group(2)
+        second_part_of_license_type = match.group(1).upper()
+        if CREATIVE_COMMONS_PATTERN.search(url):
+            first_part_of_license_type = "CC"
+        else:
+            raise Exception()
```

Comment:
I have few questions:

1. why we want to completely fail?
2. If yes, shouldn't we be more specific about the error that we raising? Exception is generic.

Reply:
Since it shouldn’t have any other license, just CC-BY. If the article is not CC-BY license, it means it's not scoap3 article